src/bloom_filter/bloom_test.rs

```diff
@@ -1,56 +1,97 @@
+    // calculated from https://hur.st/bloomfilter/
+    let size_in_bytes = 1_000_000_000;
+    let expected_elements = 1_000_000_000;
+    let num_hashers = 8;
+    assert_eq!(
+        BloomFilter::prob_of_false_positive(size_in_bytes, expected_elements, num_hashers),
+        0.025_491_740_593_406_025 as f64
+    );
+    assert_eq!(
+        BloomFilter::prob_of_false_positive(1_048_576, 524288, 2),
+        0.013_806_979_447_406_826 as f64
+    )
+}
+
+#[test]
+fn bloom_suggest_size() {
+    // it's hard to derive this exactly since the algorithm is doing closest power of 2
+    // instead of exact theoretical optimum
+
+    // Define a struct to hold test case data
+    struct TestCase {
+        elements: usize,
+        fp_rate: f64,
+        expected_size: usize,
     }
-    #[test]
-    fn bloom_test_prob_of_false_positive() {
-        // calculated from https://hur.st/bloomfilter/
-        let size_in_bytes = 1_000_000_000;
-        let expected_elements = 1_000_000_000;
-        let num_hashers = 8;
+
+    // Create a vector of test cases
+    let test_cases = vec![
+        // test for minimum size
+        TestCase {
+            elements: 4_000,
+            fp_rate: 1E-7,
+            expected_size: 1024 * 1024,
+        },
+        // test for average size
+        TestCase {
+            elements: 1_000_000,
+            fp_rate: 1E-4,
+            expected_size: 4_194_304,
+        },
+        // Add more test cases here as needed
+    ];
+
+    for test_case in test_cases {
+        let tested_size = BloomFilter::suggest_size_in_bytes(test_case.elements, test_case.fp_rate);
+        let simplified_size = simplified_suggest_size(test_case.elements, test_case.fp_rate);
+
         assert_eq!(
-            BloomFilter::prob_of_false_positive(size_in_bytes, expected_elements, num_hashers),
-            0.025_491_740_593_406_025 as f64
+            tested_size, test_case.expected_size,
+            "Failed for elements: {}, fp_rate: {}",
+            test_case.elements, test_case.fp_rate
         );
         assert_eq!(
-            BloomFilter::prob_of_false_positive(1_048_576, 524288, 2),
-            0.013_806_979_447_406_826 as f64
-        )
-    }
-
-    #[test]
-    fn bloom_suggest_size() {
-        // it's hard to derive this exactly since the algorithm is doing closest power of 2
-        // instead of exact theoretical optimum
-        let expected_elements = 1_000_000;
-        let desired_false_positive_rate = 0.0001 as f64;
-        let theoretical_optimum = ((expected_elements as f64 * desired_false_positive_rate.ln())
-            / f64::ln(1.0 / 2.0f64.powf(2.0f64.ln())))
-        .ceil()
-        .div_euclid(8f64) as usize;
-        let suggested_size =
-            BloomFilter::suggest_size_in_bytes(expected_elements, desired_false_positive_rate);
-        assert_eq!(suggested_size, 4_194_304);
-        assert_eq!(suggested_size, theoretical_optimum.next_power_of_two())
+            tested_size, simplified_size,
```

Comment:
If `simplified_suggest_size` is an improved form of `suggest_size_in_bytes`, then it is better not to have it in the unit tests like this. Instead, the implementation of `suggest_size_in_bytes` should be replaced with the implementation of `simplified_suggest_size` (probably in a separate change).

Reply:
i generally agree.